### ISSUE TITLE
if legacy decorators return a descriptor, [[Define]] semantics are used despite class-properties loose mode

### ISSUE THREAD

**Issue by trusktr** [babel/babel#12040]
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**

Even if `loose: true` is specified for proposal-class-properties, [[Define]] semantics are still used for class fields when legacy decorators return a descriptor.

**Input Code**

...

**Expected behavior**

The class fields should always use `[[Set]]` semantics (this.foo = 123) regardless of what the decorators return. (Maybe this is not correct: I'm coming from TypeScript (with `useDefineForClassFields: false`, and decorators from TS code break when compiled with Babel).

**Babel Configuration**

Just class-properties with loose:true, and decorators with legacy:true.

- Babel version(s): 7.11

**Comment by sorvell** [comment]
This appears to happen even if just the "@babel/plugin-proposal-decorators" plugin is included and no decorator is used.

Given a `.babelrc` of:

```js
{
  "plugins":[
    ["@babel/plugin-proposal-decorators", {"decoratorsBeforeExport": true}],
    ["@babel/plugin-proposal-class-properties", {"loose": true}]
    ]
}
```

Input:

```js
class A {
  prop = true;
}
```

The output unexpectedly still uses defineProperty...

```js
class A {
  constructor() {
    _defineProperty(this, "prop", true);
  }
}
```

**Comment by nicolo-ribaudo** [comment]
@sorvell I'm not sure about why, but switching the plugins fixes the output.

**Comment by sorvell** [comment]
Wow, you're right. I can confirm that.

I believe that's counter to the documentation [here](https://babeljs.io/docs/en/babel-plugin-proposal-decorators#legacy).

**Comment by sorvell** [comment]
Hmm, I still may be hitting the issue described here. Will provide more info.

**Comment by nicolo-ribaudo** [comment]
Yes, since the non-legacy decorators version implemented by Babel is fully descriptor-based, there is no way to avoid `defineProperty` when decorating a class.

**Comment by sorvell** [comment]
This seems to occur also with the legacy version of decorators as noted in the original issue description.

**Comment by agileago** [comment]
keep in mind. is there a way to keep consist with typescript `useDefineForClassFields: false` behavior ?
